bio_037_chl_a/contents/src/__init__.py

```diff
@@ -14,309 +14,479 @@
-        # nodata value -5 equals 251 for Byte type?
-        cmd = ['gdal_translate','-q', '-a_nodata', str(NODATA_VALUE), '-a_srs', 'EPSG:4326', sds_path, tif] #'-q' means quite so dont see it
+        # generate a name to save the tif file we will translate the netcdf file into
+        tif = '{}.tif'.format(os.path.splitext(f)[0])
+        # tranlate the netcdf into a tif
+        cmd = ['gdal_translate','-q', '-a_nodata', str(NODATA_VALUE), '-a_srs', 'EPSG:4326', sds_path, tif] 
         logging.debug('Converting {} to {}'.format(f, tif))
-        subprocess.call(cmd) #using the gdal from command line from inside python
+        # using gdal from command line from inside python
+        subprocess.call(cmd) 
+        # add the new tif files to the list of tifs
         tifs.append(tif)
     return tifs
 
-
-
-
-
-def fetch(new_dates):
-
-    # 2. Loop over the new dates, check if there is data available, and attempt to download the hdfs
+def fetch(dates):
+    '''
+    Fetch files by datestamp
+    INPUT   dates: list of dates we want to try to fetch, in the format YYYYMMDD (list of strings)
+    RETURN  files: list of file names for netcdfs that have been downloaded (list of strings)
+    '''
+    # make an empty list to store names of the files we downloaded
     files = []
-    for date in new_dates:
-        # Setup the url of the folder to look for data, and the filename to download to if available
+    # go through each input date
+    for date in dates:
+        # get the url to download the file from the source for the given date
         url = getUrl(date)
-        #starts as string, strptime changes to datetime object, strfttime reformats into string)
+        # get the filename we want to save the file under locally
         f = getFilename(date)
+        logging.debug('Fetching {}'.format(url))
         try:
+            # try to download the data
             urllib.request.urlretrieve(url, f)
+            # if successful, add the file to the list of files we have downloaded
             files.append(f)
             logging.info('Successfully retrieved {}'.format(f))# gives us "Successully retrieved file name"
 
         except Exception as e:
-            logging.info('Unable to retrieve data from {}, most likely NASA has not uploaded file'.format(url))
-            #NASA does not upload the previous month's chlorophyll until the middle of the next month
-            #Error is raised when trying to access this file via URL as the file has not been uploaded by NASA
-            #Send this error to log file instead of paper trails
+            # if unsuccessful, log that the file was not downloaded
+            # NASA does not upload the previous month's chlorophyll until the middle of the next month
+            # error is raised when trying to access this file via URL as the file has not been uploaded by NASA
+            logging.info('Unable to retrieve data from {}, most likely NASA has not uploaded file'.format(url))A
```

Comment:
why did you comment out this info?

Reply:
I deleted it because there is another log right below it saying the same thing, so they were duplicative. I kept the error log (rather than the info log) because when you use logging.error(), we get an email notification with the error message (so I will know that it didn't work).